File: lib/shared/utils/catalog_slug.dart
/// Normalizes catalog values to stable filter IDs (matches backend catalog-slug).
String normalizeCatalogSlug(String raw) {
  return raw
      .trim()
      .toLowerCase()
      .replaceAll(RegExp(r'[^a-z0-9]+'), '_')
      .replaceAll(RegExp(r'^_|_$'), '');
}

bool catalogSlugMatches(String value, String slug) {
  return normalizeCatalogSlug(value) == slug;
}

File: backend/src/lib/catalog-slug.ts
/** Normalizes free-text catalog values to stable filter IDs (e.g. "HIV/AIDS" → "hiv_aids"). */
export function toCatalogSlug(raw: string): string {
  return raw
    .trim()
    .toLowerCase()
    .replace(/[^a-z0-9]+/g, '_')
    .replace(/^_|_$/g, '');
}
